documentation/latest/intro/concepts.md

```diff
@@ -0,0 +1,144 @@
+connection, where we have two kinds of certificates.
+
+- **Register certificate**: Is the certificate that the Edge Admin or factory
+  admin injects for the first time in the edge device. This device can only make
+  a registration Request to the Fleet management API.
+- **User certificate**: Is the certificate that it's only valid for the specific
+  device. This certificate uses a key that shoudln't left the device, so the
+  API signs the certificate with the User Pub Key.
+
+Communication from Edgedevice to the API is like this:
+
+{% mermaid %}
+sequenceDiagram
+autonumber
+participant Agent as Device Agent
+participant YGGD as Yggdrasil
+participant API as Fleet Managment API
+
+rect rgb(226, 232, 238)
+  Agent ->> YGGD: Register (GRPC+Unixsocket)
+  YGGD ->> Agent: 200 OK
+  Agent ->> YGGD: Send with registration directive
+  YGGD ->> API: Data message with registration directive
+  API ->> YGGD: 200 OK with signed certificate
+  YGGD ->> Agent: 200 OK with signed certificate.
+  Note over YGGD, Agent: New client certificate writen
+end
+
+rect rgb(233, 237, 202)
+  loop Heartbeat
+    Agent -->> YGGD: Heartbeat
+    YGGD-->> API: Heartbeat
+    API -->> YGGD: 200 OK
+    YGGD-->> Agent: 200 OK
+  end
+end
+{% endmermaid %}
+
+1. Flotta device-agent(Agent) register to Yggdrasil using Unixsocket and GRPC
+   connection
+2. Yggdrasil allows device-worker to register in the platform.
+3. Agent needs to register to the API, and for that sends a data-message with
+   the registration directive that includes a Certificate Signed Request(CSR).
+4. Yggdrasil forward that message in the configured transport to the Operator API
+5. If registration is successful, it will return a signed certificate based on
+   the CSR
+6. Yggdrasil sends to the agent the response. Agent rewrite the client certs to
+   create new connections with the device certificates.
+7. -10 Each minute, Agent sends heartbeat data to the operator, if the operator
+returns 200 OK nothing happens, if returns 401 for the certificate expired the
+registration process will be started again.
+
+## EdgeDevice
+
+An Edgedevice can be multiple things, from a tiny RPI Zero to a big X86 server.
+For flotta, an Edgedevie is an Operating system where device-agent and Yggdrasil
+are running, and some container workloads can be run by this daemon.
+
+The communication from the edgedevice to the Kubernetes server is not required,
+so all communication is to the Fleet management API. At the same time, there are
+no things like CNI, CSI interfaces for getting it lightweight.
+
+What project-flotta focussed is that edgedevices can have one or multiple specs:
+
+- Networks can be flaky, like a satellite, NB-IoT or Lorawan.
+- Power consumption is a high priority; these devices can run on batteries.
+- Devices can be turned off for a long period.
+- Devices are insecure by default.
+- Communication should always be encrypted.
+
+### Yggdrasil
```

Comment:
Should we link to the project?

Reply:
Is the concept page, that is located on system-req and on the dev guide.